### ISSUE TITLE
Feature request: Add on_enabled() to chests

### ISSUE THREAD

**Issue by mattias-p** [christopho/solarus#898]
In Tunics we make chests appear when you press certain buttons. Sometimes enemies who happen to be in the wrong place at the wrong time get stuck in these chests. A better behavior would be to push away or kill the enemies that would get stuck in the chest at the time when it appears. To script this I believe you'd need a new `chest:on_enabled()` method.

**Comment by Renkineko** [comment]
You could maybe check in lua if there is some enemies in the region of the chest (or colliding the chest) before it appears? Don't know if it's possible, but with the function method entity:overlaps ( http://www.solarus-games.org/doc/latest/lua_api_entity.html#lua_api_entity_overlaps) ) i think it could be possible (maybe after the chest appears to be sure the overlaps is detected by the engine)

**Comment by mattias-p** [comment]
That works except for the timing. If the hero overlaps with the appearing chest, the appearance is deferred until the hero moves out of the way, and so the check must be deferred as well. I'm not sure if that's the only case when chest appearance is delayed.

The timing problem could be worked around by setting a frequent periodic timer to look for the chest's appearance. In the common case my guess is it would have to look only once.
